### PR TITLE
Fixes #35403 - disable public_file_server in production

### PR DESCRIPTION
**Pull request by evgeni**
Depends on https://github.com/theforeman/puppet-foreman/pull/1078

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@
   # config.action_dispatch.rack_cache = true
 
   # Disable Rails's static asset server (Apache or nginx will already do this).
```
Comment (ekohl) [inline config/environments/production.rb:18]: I love comments that are completely opposite to the actual line below it ...

**Review by ekohl — APPROVED** [review]
:+1: pending the installer changes.

File: config/environments/production.rb (hunk 1 above)
Comment (evgeni) [reply config/environments/production.rb:18]: Yeah, I looked for the history of that line, but it was already out of sync with the comment when originally committed by Ohad:

https://github.com/theforeman/foreman/commit/017e1049b87d914af69a8eba8ce18720b1304357#diff-da60b4e96eff2b132991226d308949e23f4ef3aad45ad59edd09cbc32cc6251e

**Comment by evgeni** [comment]
The installer changes have been merged (https://github.com/theforeman/puppet-foreman/pull/1078) and nightly remained green after that.

Smoker tests, that assert no errors are seen in the Browser console, also show no (new) failures (Foreman RPM is green, Foreman DEB has unrelated errors about "Scrollbar test exception: TypeError" which seem harmless so far, Katello has the one error about Ansible collections yielding the "wrong" page which is not really wrong, just an unexpected URL).

I am going to merge this, to ensure we actually see errors if stuff is still somewhere served from Rails.

**Pull request merged**